lib/segment/src/index/hnsw_index/graph_layers.rs

```diff
@@ -124,55 +124,33 @@ pub trait GraphLayersBase {
         points_scorer: &mut FilteredScorer,
         is_stopped: &AtomicBool,
     ) -> CancellableResult<ScoredPointOffset> {
-        let mut links: Vec<PointOffsetType> = Vec::with_capacity(2 * self.get_m(0));
-
+        let mut links_buffer = Vec::new();
```

Comment:
Why not pre-allocating is better?

Reply:
Because this function is a wrapper now, instead of greedy search, we do a call of nested `search_entry_on_level`, where we do preallocation: `links.reserve(limit);`.
But I see the preallocated size is different now.